From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kyle Smith <kyleafmine@gmail.com>
Date: Mon, 22 Apr 2024 03:21:57 -0300
Subject: [PATCH] API: Add PlayerConfigurationEvent

An event that's fired when a player connection enters the 'configuring' state.

diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerConfigurationEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerConfigurationEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..aec065bd577aac2b7436b3ca59a34a5923679a33
--- /dev/null
+++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerConfigurationEvent.java
@@ -0,0 +1,34 @@
+package com.destroystokyo.paper.event.player;
+
+import org.bukkit.entity.Firework;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.player.PlayerEvent;
+import org.bukkit.inventory.EquipmentSlot;
+import org.bukkit.inventory.ItemStack;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Fired when a player's connection enters the configuration state for any reason
+ */
+public class PlayerConfigurationEvent extends PlayerEvent {
+
+    private static final HandlerList HANDLER_LIST = new HandlerList();
+
+    public PlayerElytraBoostEvent(@NotNull Player player) {
+        super(player);
+    }
+
+    @NotNull
+    @Override
+    public HandlerList getHandlers() {
+        return HANDLER_LIST;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return HANDLER_LIST;
+    }
+}
